great_expectations/profile/basic_dataset_profiler.py

```diff
@@ -446,9 +448,11 @@ def _create_expectations_for_datetime_column(cls, dataset, column):
                 max_value = max_value + datetime.timedelta(days=365)
             except OverflowError as o_err:
                 max_value = datetime.datetime.max
+            except TypeError as o_err:
+                max_value = parse(max_value) + datetime.timedelta(days=365)
 
         if min_value is not None or max_value is not None:
-            dataset.expect_column_values_to_be_between(column, min_value=min_value, max_value=max_value)
+            dataset.expect_column_values_to_be_between(column, min_value, max_value, parse_strings_as_datetimes=True)
```

Comment:
why parse strings as datetimes here?

Reply:
```if (isinstance(val, string_types) != isinstance(min_value, string_types)) or (isinstance(val, string_types) != isinstance(max_value, string_types)):
                        raise TypeError(
                            "Column values, min_value, and max_value must either be None or of the same type.")```

Above exception was always being thrown if column was datetime-like type because min_value/max_value was always being converted to string when `dataset.expect_column_values_to_be_between` is called (from step that makes args json-serializable)